src/work-allocation-2/containers/task-assignment-confirm/task-assignment-confirm.component.ts

```diff
@@ -58,11 +71,12 @@ export class TaskAssignmentConfirmComponent implements OnInit {
   }
 
   public onCancel(): void {
-    this.router.navigate([this.rootPath, 'my-work', 'list']);
+    // Use returnUrl to return the user to the "All work" or "My work" screen, depending on which one they started from
+    this.router.navigate([this.returnUrl]);
   }
 
   private reportSuccessAndReturn(): void {
-    const message = InfoMessage.REASSIGNED_TASK;
+    const message = this.verb === 'Assign' ? InfoMessage.ASSIGNED_TASK : InfoMessage.REASSIGNED_TASK;
```

Comment:
Again, calculating verbs makes me feel queasy.

Reply:
@paulhowes-HMCTS - as above, happy to look at an alternative approach.

Reply:
@paulhowes-HMCTS - are you happy to resolve this conversation, given we discussed that here the verb is used as a discriminator to select the correct message from the `InfoMessage` enum?

Reply:
Resolved!